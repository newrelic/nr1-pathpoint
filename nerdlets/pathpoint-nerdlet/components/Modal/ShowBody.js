import React, { Component } from 'react';
import PropTypes from 'prop-types';

// IMPORT ICONS
import graphImage from '../../images/graph.png';

// IMPORT MESSAGES
import messages from '../../config/messages.json';

// IMPORT COMPONENTS
import { BodyTuneFormModal } from './TuneFormModal';
import { BodyLogoFormModal } from './LogoFormModal';
import { BodyQueryFormModal } from './QueryFormModal';
import { BodyFlameFormModal } from './FlameFormModal';
import { BodyCanaryFormModal } from './CanaryFormModal';
import { BodySupportFormModal } from './SupportFormModal';
import { BodyFileErrorFormModal } from './FileErrorFormModal';
import { BodyJsonConfigurationFormModal } from './JsonConfigurationFormModal';
import { BodyGeneralConfigurationFormModal } from './GeneralConfigurationFormModal';
import { BodyBackgroundProcessesFormModal } from './BackgroundProcessesFormModal';

export default class ShowBody extends Component {
  constructor(props) {
    super(props);
    this.state = {
      url: '',
      text: '',
      type: '',
      subject: '',
      name: '',
      company: '',
      account: '',
      email: '',
      phone: '',
      message: '',
      min_count: 0,
      max_count: 0,
      min_apdex: 0,
      max_response_time: 0,
      max_error_percentage: 0,
      max_avg_response_time: 0,
      max_total_check_time: 0,
      min_success_percentage: 0
    };
  }

  componentDidMount() {
    const { stageNameSelected } = this.props;
    if (
      stageNameSelected &&
      stageNameSelected.datos &&
      stageNameSelected.datos[0]
    ) {
      let {
        min_count,
        max_count,
        min_apdex,
        max_response_time,
        max_error_percentage,
        max_avg_response_time,
        max_total_check_time,
        min_success_percentage
      } = this.state;
      switch (stageNameSelected.datos[0].type) {
        case 'PRC':
        case 'PCC':
          min_count = stageNameSelected.datos[0].min_count;
          max_count = stageNameSelected.datos[0].max_count;
          break;
        case 'APP':
        case 'FRT':
          min_apdex = stageNameSelected.datos[0].min_apdex;
          max_response_time = stageNameSelected.datos[0].max_response_time;
          max_error_percentage =
            stageNameSelected.datos[0].max_error_percentage;
          break;
        case 'SYN':
          max_avg_response_time =
            stageNameSelected.datos[0].max_avg_response_time;
          max_total_check_time =
            stageNameSelected.datos[0].max_total_check_time;
          min_success_percentage =
            stageNameSelected.datos[0].min_success_percentage;
      }
      this.setState({
        min_count: min_count,
        max_count: max_count,
        min_apdex: min_apdex,
        max_response_time: max_response_time,
        max_error_percentage: max_error_percentage,
        max_avg_response_time: max_avg_response_time,
        max_total_check_time: max_total_check_time,
        min_success_percentage: min_success_percentage
      });
    }
    /* istanbul ignore next */
    if (messages.configuration.support.options_select_support_02.service_1) {
      this.setState({
        subject:
          messages.configuration.support.options_select_support_02.service_1
      });
    }
  }

  handleOnChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  handleSubmitSupport = event => {
    event.preventDefault();
    const {
      subject,
      name,
      company,
      account,
      email,
      phone,
      message
    } = this.state;
    const { handleSaveUpdateSupport } = this.props;
    handleSaveUpdateSupport({
      subject,
      name,
      company,
      account,
      email,
      phone,
      message
    });
  };

  handleSubmitLogo = event => {
    event.preventDefault();
    const { url, text, type } = this.state;
    const { LogoFormSubmit, _onClose } = this.props;
    LogoFormSubmit({ url, text, type }, _onClose);
  };

  handleSubmitTune = event => {
    event.preventDefault();
    const {
      min_count,
      max_count,
      min_apdex,
      max_response_time,
      max_error_percentage,
      max_avg_response_time,
      max_total_check_time,
      min_success_percentage
    } = this.state;
    const { handleSaveUpdateTune } = this.props;
    handleSaveUpdateTune({
      min_count,
      max_count,
      min_apdex,
      max_response_time,
      max_error_percentage,
      max_avg_response_time,
      max_total_check_time,
      min_success_percentage
    });
  };

  showBodyRender = () => {
    const { type } = this.state;
    const { viewModal } = this.props;
    switch (viewModal) {
      case 0:
        return <img src={graphImage} />;
      case 1:
        return <BodyQueryFormModal {...this.props} />;
      case 2:
        return (
          <BodyTuneFormModal
            {...this.props}
            handleOnChange={this.handleOnChange}
            handleSubmitTune={this.handleSubmitTune}
          />
        );
      case 3:
        return <div />;
      case 4:
        return <BodyJsonConfigurationFormModal {...this.props} />;
      case 5:
        return (
          <BodySupportFormModal
            {...this.props}
            handleOnChange={this.handleOnChange}
            handleSubmitSupport={this.handleSubmitSupport}
          />
        );
      case 6:
        return <BodyCanaryFormModal {...this.props} />;
      case 7:
        return <BodyFlameFormModal {...this.props} />;
      case 8:
        return <BodyFileErrorFormModal {...this.props} />;
      case 9:
        return (
          <BodyGeneralConfigurationFormModal
            {...this.props}
            handleOnChange={this.props.HandleCredentialsFormChange}
            handleFormSubmit={this.props.handleSaveUpdateGeneralConfiguration}
            resetCredentials={this.props.resetCredentials}
          />
        );
      case 10:
        return (
          <BodyLogoFormModal
            handleSubmitLogo={this.handleSubmitLogo}
            handleOnChange={this.handleOnChange}
            type={type}
          />
        );
      case 11:
        return <BodyBackgroundProcessesFormModal {...this.props} />;
    }
  };

  render() {
    return this.showBodyRender();
  }
}

ShowBody.propTypes = {
  viewModal: PropTypes.number.isRequired,
  LogoFormSubmit: PropTypes.func.isRequired,
  _onClose: PropTypes.func.isRequired,
  handleSaveUpdateTune: PropTypes.func.isRequired,
  handleSaveUpdateSupport: PropTypes.func.isRequired,
  HandleCredentialsFormChange: PropTypes.func,
  resetCredentials: PropTypes.func,
  stageNameSelected: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  handleSaveUpdateGeneralConfiguration: PropTypes.func
};
