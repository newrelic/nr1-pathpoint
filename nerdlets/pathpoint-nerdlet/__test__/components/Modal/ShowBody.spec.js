import React from 'react';
import { create } from 'react-test-renderer';
import ShowBody from '../../../components/Modal/ShowBody';

const accountIDs = [
  {
    name: 'WigiBoards',
    id: 2710112
  },
  {
    name: 'Account 2',
    id: 7859641
  },
  {
    name: 'Account 3',
    id: 7859642
  },
  {
    name: 'Account 4',
    id: 7859642
  }
];

describe('ShowBody component', () => {
  test('ShowBody with default data in 0 view', () => {
    const body = create(
      <ShowBody
        handleContinueUAMButton={jest.fn()}
        GetCurrentHistoricErrorScript={jest.fn()}
        GetCurrentConfigurationJSON={jest.fn()}
        SetConfigurationJSON={jest.fn()}
        validateKpiQuery={{}}
        stageNameSelected={{
          selectedCase: 0,
          datos: [
            {
              type: 'SYN',
              query:
                "SELECT filter(percentage(count(result),WHERE result='SUCCESS'),WHERE 1=1) as success, max(duration) as duration, max(longRunningTasksAvgTime) as request from SyntheticCheck,SyntheticRequest WHERE monitorName='BDB Live person'",
              max_avg_response_time: 0.7,
              max_total_check_time: 1.25,
              min_success_percentage: 98,
              success_percentage: 0,
              max_duration: 0,
              max_request_time: 0
            }
          ]
        }}
        ValidateIngestLicense={jest.fn()}
        ValidateUserApiKey={jest.fn()}
        ToggleEnableSubmit={jest.fn()}
        installUpdateBackgroundScripts={jest.fn()}
        disableGeneralConfigurationSubmit={false}
        credentialsData={{
          accountId: null,
          ingestLicense: null,
          userAPIKey: null,
          dropTools: null,
          flameTools: null,
          loggin: null
        }}
        licenseValidations={{
          ingestLicense: false,
          userApiKey: false
        }}
        UpdateJSONMetaData={jest.fn()}
        jsonMetaData={{
          description: '',
          note: ''
        }}
        GetHistoricJSONData={jest.fn().mockReturnValue([])}
        JSONModal={{
          view: 0,
          historic: []
        }}
        UpdateItemSelectFromHistoric={jest.fn()}
        currentHistoricSelected={null}
        RestoreJSONFromHistoric={jest.fn()}
        username="PathPoint"
        chargueSample={jest.fn()}
        testQuery={jest.fn()}
        handleSaveUpdateQuery={jest.fn()}
        handleSaveUpdateSupport={jest.fn()}
        testText="Bad query"
        resultsTestQuery={{ type: 'default' }}
        goodQuery={false}
        modifiedQuery
        handleChangeTexarea={jest.fn()}
        viewModal={4}
        width="230px"
        handleSaveUpdateTune={jest.fn()}
        _onClose={jest.fn()}
        LogoFormSubmit={jest.fn()}
        accountIDs={accountIDs}
        HandleCredentialsFormChange={jest.fn()}
        handleSaveUpdateGeneralConfiguration={jest.fn()}
        resetCredentials={jest.fn()}
      />
    );
    expect(body.toJSON()).toMatchSnapshot();
  });
});
