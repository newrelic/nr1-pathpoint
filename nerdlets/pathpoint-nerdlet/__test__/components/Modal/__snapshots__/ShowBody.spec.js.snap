// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowBody component ShowBody with default data in 0 view 1`] = `
<div
  style={
    Object {
      "width": "400px",
    }
  }
>
  <div>
    <div
      style={
        Object {
          "marginBottom": "20px",
        }
      }
    >
      <label>
        Description: 
      </label>
      <input
        className="text-input-custom"
        maxLength={64}
        onChange={[Function]}
        type="text"
      />
    </div>
    <div>
      <label>
        Note: 
      </label>
      <textarea
        className="text-input-custom"
        onChange={[Function]}
        rows="5"
        type="text"
      />
    </div>
    <div
      style={
        Object {
          "marginTop": "10px",
        }
      }
    >
      <p>
        User:
        <strong>
           PathPoint
        </strong>
      </p>
    </div>
  </div>
  <div
    id="upload-file-submit"
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
        "marginTop": "30px",
      }
    }
  >
    <button
      className="btn btn-default"
      color="primary"
      disabled={true}
      onClick={[Function]}
      style={
        Object {
          "background": "#0178bf",
          "color": "white",
          "marginRight": "20px",
        }
      }
      type="submit"
      variant="contained"
    >
      <img
        src="SvgrURL"
        style={
          Object {
            "height": "auto",
            "marginRight": "4px",
            "width": "16px",
          }
        }
      />
      From File
    </button>
    <div
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <button
        className="btn btn-default"
        color="primary"
        disabled={false}
        onClick={[MockFunction]}
        style={
          Object {
            "background": "#0178bf",
            "color": "white",
          }
        }
        type="button"
        variant="contained"
      >
        <img
          src="SvgrURL"
          style={
            Object {
              "marginRight": "4px",
            }
          }
        />
        From History
      </button>
    </div>
  </div>
  <input
    accept=".json"
    id="file-upload"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />
</div>
`;
